cuqi/distribution/_distribution.py

```diff
@@ -320,8 +320,13 @@ def _condition(self, *args, **kwargs):
 
                 elif len(var_args)>0:                      #Some keywords found
                     # Define new partial function with partially defined args
-                    func = partial(var_val, **var_args)
-                    setattr(new_dist, var_key, func)
+                    # Try evaluating function partially with keywords
```

Comment:
We will assume that calling succcessfully the object with only some of the required arguments will only work if the outcome is something that represents a partial evaluation. We could consider being more explicit here and requesting all elements to have a "evaluate" method or something like that, rather than "call".

Reply:
See also #183